controller/konnect/reconciler_generic.go

```diff
@@ -526,13 +525,20 @@ func (r *KonnectEntityReconciler[T, TEnt]) Reconcile(
 	}
 
 	if res, err := ops.Update[T, TEnt](ctx, sdk, r.SyncPeriod, r.Client, ent); err != nil {
+		ent.GetKonnectStatus().ServerURL = apiAuth.Spec.ServerURL
+		ent.GetKonnectStatus().OrgID = apiAuth.Status.OrganizationID
+		if errUpd := r.Client.Status().Update(ctx, ent); errUpd != nil {
+			if k8serrors.IsConflict(errUpd) {
+				return ctrl.Result{Requeue: true}, nil
+			}
+			return ctrl.Result{}, fmt.Errorf("failed to update in cluster resource after Konnect update: %w %w", errUpd, err)
+		}
+
 		return ctrl.Result{}, fmt.Errorf("failed to update object: %w", err)
 	} else if !res.IsZero() {
 		return res, nil
 	}
 
-	ent.GetKonnectStatus().ServerURL = apiAuth.Spec.ServerURL
-	ent.GetKonnectStatus().OrgID = apiAuth.Status.OrganizationID
```

Comment:
Shouldn't we leave them here as well so the `ServerURL` and `OrgID` are populated on the entity update success? 🤔

Reply:
Yes, I'm not sure why I removed this here 🤔 

a3f201ff5ebf7535c2454bf6151ad4cf2b357759